multitemporal/bin/passthrough.pyx

```diff
@@ -7,37 +7,19 @@ cimport cython
 @cython.cdivision(True)
 
 def get_nout(int nin, np.ndarray[np.float32_t, ndim=1, negative_indices=False] params not None):
-    return nin
-    
-def passthrough(np.ndarray[np.float32_t, ndim=3, negative_indices=False] data not None,
-                float missingval,
-                np.ndarray[np.float32_t, ndim=1, negative_indices=False] params not None):
-
-    return data
+    """Return the expected number of time periods (each is a year).
 
+    This is based on the input number of time periods and the params of the
+    operation.
     """
-    cdef unsigned int nfr = data.shape[0]
-    cdef unsigned int nyr = data.shape[1]
-    cdef unsigned int npx = data.shape[2]
-
-    cdef float nothing = params[0]
-
-    cdef int nin = get_nout(nfr, params)
-    cdef np.ndarray[np.float32_t, ndim=3] result = np.zeros((nin,nyr,npx), dtype='float32')
-
-    cdef unsigned int i,j,k
+    return nin
 
-    for k in range(npx):
-        for j in range(nyr):
-            for i in range(nfr):
-                if i+1 < minframe or i+1 > maxframe:
-                    continue
-                if data[i,j,k] != missingval and data[i,j,k] > thresh:
-                    result[0,j,k] = result[0,j,k] + data[i,j,k]
-                    result[1,j,k] = result[1,j,k] + 1.0
-            if result[1,j,k] == 0.0:
-                result[0,j,k] = missingval
-                result[1,j,k] = 0.0
+# impossible to specify an unknown number of array dimensions at compile time:
+#                          vvvvvv
+# np.ndarray[np.float32_t, ndim=3, negative_indices=False] data not None
+def passthrough(data not None, ignored_missing_out, ignored_params):
+    """Identity function as a multitemporal module.
 
-    return result
+    Useful for testing and for using input data as output data.
     """
+    return data
```

Comment:
So I realize that the looping here was not necessary, but I wonder if the returning a copy of the input data object (which points at the same data array) is problematic for multitemporal's chaining?  @bhbraswell do you smell any issue with that?

Reply:
I don't think it would be a problem in any cases I can think of.